week-1/implement/dedupe.js

```diff
@@ -1 +1,14 @@
-function dedupe() {}
+// function dedupe() {}
+
+//............................................
+
+//ANSWER
+
+function dedupe(array) {
+    const uniqueSet = new Set(array);
+    const dedupedArray = [...uniqueSet];
```

Comment:
Great work mate, but I have a question here. Do you know whats line 9 doing  I mean this bit `[...uniqueSet]`. If yes ca you explain in a comment please?

Reply:
> [...uniqueSet]

The "... spread operator."  In this case [...uniqueSet], is used to spread the elements of uniqueSet into a new array.
The spread operator takes the individual elements from the uniqueSet and spreads them out, element by element, into a new array.
So, [...uniqueSet] is a concise way of creating a new array with all the elements from uniqueSet, making it easier to work with arrays and eliminate duplicates.